packages/ses/src/compartment.js

```diff
@@ -227,21 +208,36 @@ export const makeCompartmentConstructor = (
     // moduleMap can be invalid in ways that cannot be detected in the
     // constructor, but these checks allow us to throw early for a better
     // developer experience.
-    for (const [specifier, aliasNamespace] of entries(moduleMap || {})) {
-      if (typeof aliasNamespace === 'string') {
+    for (const [specifier, moduleDescriptor] of entries(moduleMap || {})) {
+      if (typeof moduleDescriptor === 'string') {
         // TODO implement parent module record retrieval.
         throw TypeError(
           `Cannot map module ${q(specifier)} to ${q(
-            aliasNamespace,
+            moduleDescriptor,
           )} in parent compartment`,
         );
-      } else if (weakmapGet(moduleAliases, aliasNamespace) === undefined) {
-        // TODO create and link a synthetic module instance from the given
-        // namespace object.
-        throw ReferenceError(
-          `Cannot map module ${q(
-            specifier,
-          )} because it has no known compartment in this realm`,
+      } else if (isObject(moduleDescriptor)) {
+        if (weakmapGet(moduleAliases, moduleDescriptor) !== undefined) {
+          // No furthere validation for module namespace objects.
+          // They do bear inspection before evaluation gracefully.
```

Comment:
Did you mean?
```suggestion
          // They do bear inspection before evaluating gracefully.
```

Reply:
I think this is grammatical as written, but always down for a second or third opinion. @gibson042 how’s your grammar?

Reply:
(I went with evaluation because it agrees with inspection if that’s consequential.)

Reply:
(Pardon, don’t waste too much time on this. I delete this entire section in a planned change. I might move that up to make that clear.)

Reply:
Resolving